File: game/scripts/vscripts/invokation/game_mode/env.lua
--- Env class.
-- @classmod invokation.game_mode.Env

local class = require("pl.class")

local M = class()

--- Environment modes
-- @section modes

--- Development
-- @tfield string DEVELOPMENT
M.DEVELOPMENT = "development"

--- Production
-- @tfield string PRODUCTION
M.PRODUCTION = "production"

--- Methods
-- @section methods

--- Constructor.
-- @tparam[opt=PRODUCTION] string name Environment name.
function M:_init(name)
  self.name = name or M.PRODUCTION
  self.development = self.name == M.DEVELOPMENT
  self.production = self.name == M.PRODUCTION
end

return M

File: src/content/panorama/scripts/lib/env.ts
export enum Name {
  Development = "development",
  Production = "production",
}

export class Env {
  name: Name;

  constructor(name = isDevelopment() ? Name.Development : Name.Production) {
    this.name = name;
  }

  get development() {
    return this.name === Name.Development;
  }

  get production() {
    return this.name === Name.Production;
  }
}

function isDevelopment(): boolean {
  return Game.IsInToolsMode() || Game.GetConvarInt("developer") > 0;
}

export const ENV = new Env();
